### PR TITLE
Add process inspection for Windows

### LINKED ISSUE #1445: make our process inspection cross platform
Currently we utilize pgrep to check process state, this is only avaiable on mac and linux. We need a portable way to check process state across *nix, macOS, Windows.

The current implementation is here: https://github.com/elastic/e2e-testing/blob/master/e2e/_suites/fleet/world.go#L85-L265

To make this work we need to move this implementation into the installer internal code so that when we attach an installer we can easily inspect processes based on the underlying operating system.

@IgorGuz2000 Is this something you want to try and tackle?

### PR DESCRIPTION
**Pull request by adam-stokes**
This PR adds support for checking and verifying elastic-agent/metricbeat/filebeat processes are in their defined states on Windows. This also includes the pattern for defining new commands to be run in a platform agnostic way.

Fixes #1445  
Signed-off-by: Adam Stokes <51892+adam-stokes@users.noreply.github.com>

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests (`make unit-test`), and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: cli/config/config_test.go [hunk 1]
```diff
@@ -18,6 +18,10 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+const testResourcesBasePath = "_testresources/"
```
Comment (mdelapenya) [inline cli/config/config_test.go:21]: I think this is a leftover

File: cli/config/config_test.go [hunk 2]
```diff
@@ -18,6 +18,10 @@ import (
...
+const dockerComposeMultiple = "docker-compose-multiple.yml"
+const dockerComposeSingle = "docker-compose-single.yml"
+
```
Comment (mdelapenya) [inline cli/config/config_test.go:24]: I think this is a leftover

File: cli/config/config.go [hunk 3]
```diff
@@ -222,6 +222,12 @@ func configureLogger() {
 	}
 }
 
+type service interface{}
+type composeFile struct {
+	Version  string             `yaml:"version"`
+	Services map[string]service `yaml:"services"`
+}
+
```
Comment (mdelapenya) [inline cli/config/config.go:230]: I think this is a leftover

File: cli/config/_testresources/docker-compose-multiple.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+version: '2.3'
```
Comment (mdelapenya) [inline cli/config/_testresources/docker-compose-multiple.yml:1]: I think this is a leftover

File: cli/config/_testresources/docker-compose-single.yml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+version: '2.3'
```
Comment (mdelapenya) [inline cli/config/_testresources/docker-compose-single.yml:1]: I think this is a leftover

File: internal/deploy/remote.go [hunk 6]
```diff
@@ -59,7 +60,14 @@ func (c *remoteDeploymentManifest) ExecIn(ctx context.Context, profile ServiceRe
...
 // Inspect inspects a service
 func (c *remoteDeploymentManifest) Inspect(ctx context.Context, service ServiceRequest) (*ServiceManifest, error) {
-	return &ServiceManifest{}, nil
+	// TODO: convert to a platform agnostic command structure
+	hostname, _ := shell.Execute(ctx, ".", "powershell.exe", "hostname")
+	return &ServiceManifest{
+		Hostname:   strings.TrimSpace(hostname),
+		Connection: service.Name,
+		Alias:      service.Name,
+		Platform:   "windows",
```
[the remark this answers is not in the document]
Comment (adam-stokes) [reply internal/deploy/remote.go:69]: Yea this was what I was referring to on the call earlier, right now it's just hardcoded for windows but I have an idea how to solve this in a subsequent PR

File: go.mod [hunk 7]
```diff
@@ -6,29 +6,35 @@ require (
 	github.com/Flaque/filet v0.0.0-20201012163910-45f684403088
 	github.com/Jeffail/gabs/v2 v2.6.0
 	github.com/cenkalti/backoff/v4 v4.1.0
-	github.com/cucumber/godog v0.11.0
-	github.com/cucumber/messages-go/v10 v10.0.3
+	github.com/cucumber/godog v0.11.0 // indirect
```
Comment (mdelapenya) [inline go.mod:9]: I think this is going to conflict with #1588

Comment (adam-stokes) [reply go.mod:9]: Let's merge #1588 first then ill fix the conflicts

File: internal/installer/elasticagent_zip.go [hunk 8]
```diff
@@ -122,13 +123,22 @@ func (i *elasticAgentZIPPackage) Preinstall(ctx context.Context) error {
...
 	}
 
-	_, err = i.Exec(ctx, []string{"powershell.exe", "Expand-Archive", "-LiteralPath", binaryPath, "-DestinationPath", "C:\\", "-Force"})
-	if err != nil {
-		return err
-	}
+	output, err := i.Exec(ctx, []string{"powershell.exe", "Test-Path", "C:\\elastic-agent"})
+	log.WithFields(log.Fields{
+		"output": output,
+		"error":  err}).Trace("Checking for existence of elastic-agent installation directory")
```
Comment (mdelapenya) [inline internal/installer/elasticagent_zip.go:129]: NIT: consistent indentation
```suggestion
	log.WithFields(log.Fields{
		"output": output,
		"error":  err
	}).Trace("Checking for existence of elastic-agent installation directory")
```

**Review by mdelapenya — APPROVED** [review]

**Comment by cachedout** [comment]
Hi @adam-stokes 

This PR doesn't seem to have an issue associated with it and there's no description in the body that describes it. Could you please provide some additional context behind this change? Thanks.

**Comment by adam-stokes** [comment]
> Hi @adam-stokes
> 
> This PR doesn't seem to have an issue associated with it and there's no description in the body. Could you please provide some additional context behind this change? Thanks.

Thanks for the reminder @cachedout ! I've updated the description and included link to issue this resolves

**Pull request merged**